src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    public class Node<T> {
+        private Node<T> prev;
+        private Node<T> next;
+        private T value;
+
+        public Node(Node<T> prev, T value) {
+            this.prev = prev;
+            this.value = value;
+        }
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+
+        public Node(T value, Node<T> next) {
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value);
+        if (tail == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index can`t be:" + index
+                    + " for LinkedList size:" + size);
+        }
+        Node<T> nodeByIndex = iterateToIndex(index);
+        if (nodeByIndex == null) {
+            add(value);
+            return;
+        }
+        Node<T> newNode = new Node<>(nodeByIndex.prev, value, nodeByIndex);
+        if (head == nodeByIndex) {
+            head = newNode;
+        } else {
+            nodeByIndex.prev.next = newNode;
+        }
+        nodeByIndex.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (Object element : list) {
+            add((T) element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkOutOfMemory(index);
+        Node<T> nodeByIndex = iterateToIndex(index);
+        return (T) nodeByIndex.value;
```

Comment:
You call the checkOutOfMemory method in several places before calling the iterateToIndex method. Perhaps it is worth calling the checkOutOfMemory method in the middle of the iterateToIndex method.

Reply:
I can't because there is another check in line 47